application/src/main/java/com/ericsson/bss/cassandra/ecchronos/application/spring/RetrySchedulerService.java

```diff
@@ -0,0 +1,220 @@
+ * <p>
+ * This service periodically checks the status of nodes and attempts to reconnect based on a configurable retry policy.
+ * It uses a scheduled executor service to perform retries at fixed intervals, with the intervals and the retry logic
+ * configurable via external configurations.
+ * </p>
+ *
+ * <p>
+ * The retry logic involves calculating the delay between attempts, which increases with each subsequent retry for a node.
+ * If the maximum number of retry attempts is reached, the node is marked as unreachable.
+ * </p>
+ *
+ * <p>
+ * This service is designed to run continuously in the background, adjusting its behavior based on the state of the
+ * Cassandra cluster and the provided configurations. It also ensures that resources are properly cleaned up on shutdown.
+ * </p>
+ */
+
+@Service
+public final class RetrySchedulerService implements DisposableBean
+{
+
+    private static final Logger LOG = LoggerFactory.getLogger(RetrySchedulerService.class);
+    private static final String COLUMN_NODE_ID = "node_id";
+    private static final String COLUMN_NODE_STATUS = "node_status";
+    private static final int SCHEDULER_AWAIT_TERMINATION = 60;
+    private final EccNodesSync myEccNodesSync;
+    private final DistributedJmxConnectionProvider myJmxConnectionProvider;
+    private final DistributedNativeConnectionProvider myDistributedNativeConnectionProvider;
+    private final RetryBackoffStrategy retryBackoffStrategy;
+    private final ScheduledExecutorService myScheduler = Executors.newScheduledThreadPool(1);
+
+    public RetrySchedulerService(final EccNodesSync eccNodesSync,
+                                 final Config config,
+                                 final DistributedJmxConnectionProvider jmxConnectionProvider,
+                                 final DistributedNativeConnectionProvider distributedNativeConnectionProvider)
+    {
+        this.myEccNodesSync = eccNodesSync;
+        this.myJmxConnectionProvider = jmxConnectionProvider;
+        this.myDistributedNativeConnectionProvider = distributedNativeConnectionProvider;
+        this.retryBackoffStrategy = new RetryBackoffStrategy(config.getConnectionConfig().getJmxConnection().getRetryPolicyConfig());
+    }
+
+    @PostConstruct
+    public void startScheduler()
+    {
+        long initialDelay = retryBackoffStrategy.getInitialDelay();
+        long fixedDelay = retryBackoffStrategy.getFixedDelay();
+
+        LOG.info("Starting RetrySchedulerService with initialDelay={} ms and fixedDelay={} ms", initialDelay, fixedDelay);
+
+        myScheduler.scheduleWithFixedDelay(this::retryNodes, initialDelay, fixedDelay, TimeUnit.MILLISECONDS);
+    }
+
+    @VisibleForTesting
+    void retryNodes()
+    {
+        LOG.debug("Retrying unavailable nodes");
+        List<Node> unavailableNodes = findUnavailableNodes();
+
+        if (unavailableNodes.isEmpty())
+        {
+            LOG.info("No unavailable nodes found.");
```

Comment:
What does this mean "no unavailable nodes found" and why is it important to info log?

Reply:
it was just for manual testing! forget to remove it: